tests/test_optimize.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+from unittest import TestCase
+
+from aspire.utils import ensure
+from aspire.utils.matlab_compat import m_reshape
+from aspire.utils.optimize import conj_grad
+
+
+def identity(x):
+    return x
+
+
+def precond_fun(S, x):
```

Comment:
Where do these complicated `precond_fun` and `apply` come from? Why not just do simple matrix multiplication? It's much easier to verify what's going on there.

Reply:
Previously there is no any unit test for the CG method. I think it is good to include them with preconditioner function.  I tried to use the Cov2D case to build the tests. The data and functions are reassembly the Cov2D problem.

Reply:
The `precond_fun` was removed in the new commit 6f43523.

Reply:
Ok. I didn't say we should not test the preconditioner. I said let's not have a test case that's more complicated than necessary. A simple matrix multiplication problem *with preconditioner* will do.

Reply:
New unit tests were included in commit 6d8aeeb.

Reply:
The unit tests were simplified to use the default CG option with the `identity` preconditioner  in commit 18bbf57.